### PR TITLE
added http client connect audit event support

### PR DESCRIPTION
**Pull request by ravi-kale**
Fixes --> https://github.com/urllib3/urllib3/issues/2757

### CODE REVIEW

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -170,6 +171,9 @@ def _new_conn(self):
         if self.socket_options:
             extra_kw["socket_options"] = self.socket_options
 
+        if hasattr(sys, "audit"):
```
Comment (sethmlarson) [inline src/urllib3/connection.py:174]: This `hasattr` check gets called per connection, can we detect the feature once in the global scope and cache whether `sys.audit` is available? Let's also add a comment that mentions sys.audit being Python 3.8+

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this. Can we also add a test case that verifies the names/values of all the `sys.audit` calls for the `http` namespace?

We also want this implemented in the `main` branch, the `1.26.x` branch is no longer receiving new features.

File: src/urllib3/connection.py (hunk 1 above)
Comment (ravi-kale) [reply src/urllib3/connection.py:174]: Yeah, thanks for your feedback.
Will continue working on https://github.com/urllib3/urllib3/pull/2847. Will close this Pull Request.

**Comment by ravi-kale** [comment]
> Thanks for this. Can we also add a test case that verifies the names/values of all the `sys.audit` calls for the `http` namespace?
> 
> We also want this implemented in the `main` branch, the `1.26.x` branch is no longer receiving new features.

Hey @sethmlarson, If the `1.26.x` branch is no longer receiving a new feature, I will continue to work on my previous pull request (https://github.com/urllib3/urllib3/pull/2847)  which was closed by me.

**Pull request closed without merging**